src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -3,202 +3,422 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics.X86;
 using Internal.Runtime.CompilerServices;
 
 namespace System.Runtime.Intrinsics
 {
+    // We mark certain methods with AggressiveInlining to ensure that the JIT will
+    // inline them. The JIT would otherwise not inline the method since it, at the
+    // point it tries to determine inline profability, currently cannot determine
+    // that most of the code-paths will be optimized away as "dead code".
+    //
+    // We then manually inline cases (such as certain intrinsic code-paths) that
+    // will generate code small enough to make the AgressiveInlining profitable. The
+    // other cases (such as the software fallback) are placed in their own method.
+    // This ensures we get good codegen for the "fast-path" and allows the JIT to
+    // determine inline profitability of the other paths as it would normally.
+
     public static class Vector128
     {
         internal const int Size = 16;
 
         /// <summary>Creates a new <see cref="Vector128{Byte}" /> instance with all elements initialized to the specified value.</summary>
         /// <param name="value">The value that all elements will be initialized to.</param>
         /// <returns>A new <see cref="Vector128{Byte}" /> with all elements initialized to <paramref name="value" />.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe Vector128<byte> Create(byte value)
         {
-            var pResult = stackalloc byte[16]
+            if (Avx2.IsSupported)
             {
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-            };
+                Vector128<byte> result = CreateScalarUnsafe(value);                         // < v, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? >
```

Comment:
This is generating:
```asm
       movzx    rax, dl
       vmovd    xmm0, eax
       vpbroadcastb xmm0, xmm0
```

Given that we are calling `CreateScalarUnsafe`, I want to determine why the `movzx` is being inserted

Reply:
For the test scenario (basically when this is inlined to another method), we seem to end up shuffling the result around a bit:
```asm
       vmovd    xmm0, edi
       vpbroadcastb xmm0, xmm0
       vmovaps  xmm6, xmm0
       vmovaps  xmm0, xmm6
       vmovaps  xmm6, xmm0
```

Noting that we don't get the additional/unnecessary `movzx` here, but the output is shuffled back and forth a few times

Reply:
> I want to determine why the movzx is being inserted

I guess that is because CoreCLR promotes small integers to 32/64-bit on the evaluation stack.

Reply:
> but the output is shuffled back and forth a few times

Maybe related to https://github.com/dotnet/coreclr/issues/16619? I remember that sometimes JIT cannot eliminates pass/return-by-ref for inlined functions.

Reply:
I suspect that it *is* related to #16619. Some improvements should come with #21304, once I've worked the kinks out if it. And then there's still more work to do to get rid of all the most egregious  unnecessary copies.